OpenSim/Moco/MocoGoal/MocoContactImpulseTrackingGoal.cpp

```diff
@@ -328,43 +330,26 @@ void MocoContactTrackingGoal::calcIntegrandImpl(
 
         // Apply scale factors for this marker, if they exist.
         const auto& scaleFactorRef = m_scaleFactorRefs[ig];
-        for (int j = 0; j < 3; ++j) {
-            if (scaleFactorRef[j] != nullptr) {
-                force_ref[j] *= scaleFactorRef[j]->getScaleFactor();
-            }
+        if (scaleFactorRef != nullptr) {
+            force_ref *= scaleFactorRef->getScaleFactor();
```

Comment:
`force_ref` is type `SimTK::Vec3`, and here you are multiplying by a `double`. This should work as expected, i.e., all elements of `force_ref` are multiplied by the scalar. But it's also good to check the docs to make sure these types of operations are doing what you expect. If you wanted to be extra safe, you could re-write this section to something like this:

```c++
// Apply scale factors for this marker, if they exist.
const auto& scaleFactorRef = m_scaleFactorRefs[ig];
double scaleFactor = 1.0;
if (scaleFactorRef != nullptr) {
    scaleFactor = scaleFactorRef->getScaleFactor();
}

double error = force_model - scaleFactor * force_ref[get_impulse_axis()];
integrand += error;
```